zeebe/atomix/cluster/src/main/java/io/atomix/raft/metrics/RaftRoleMetrics.java

```diff
@@ -16,86 +16,77 @@
  */
 package io.atomix.raft.metrics;
 
-import io.prometheus.client.Counter;
-import io.prometheus.client.Gauge;
-import io.prometheus.client.Histogram;
+import static io.atomix.raft.metrics.RaftRoleMetricsDoc.*;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class RaftRoleMetrics extends RaftMetrics {
 
-  private static final Gauge ROLE =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("role")
-          .help("Shows current role")
-          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
-          .register();
-
-  private static final Counter HEARTBEAT_MISS =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("heartbeat_miss_count")
-          .help("Count of missing heartbeats")
-          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
-          .register();
-
-  private static final Histogram HEARTBEAT_TIME =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("heartbeat_time_in_s")
-          .help("Time between heartbeats")
-          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
-          .register();
-  private static final Gauge ELECTION_LATENCY =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("election_latency_in_ms")
-          .help("Duration for election")
-          .labelNames(PARTITION_GROUP_NAME_LABEL, PARTITION_LABEL)
-          .register();
-
-  private final Gauge.Child role;
-  private final Counter.Child heartbeatMiss;
-  private final Histogram.Child heartbeatTime;
-  private final Gauge.Child electionLatency;
-
-  public RaftRoleMetrics(final String partitionName) {
+  private final Counter heartbeatMiss;
+  private final Timer heartbeatTime;
+  private final AtomicLong roleValue = new AtomicLong(0L);
```

Comment:
:upside_down_face: Can probably be an `AtomicInteger` (in fact, it should be an `EnumMeter`, but I guess this might break compat)

Reply:
Yeah, it cannot be an EnumMeter because the encoding is different. 
Will change to `AtomicInteger`  :+1: